src/fides/cli/commands/util.py

```diff
@@ -38,69 +39,23 @@ def init(ctx: click.Context, fides_directory_location: str) -> None:
     """
 
     executed_at = datetime.now(timezone.utc)
-    separate = lambda: print("-" * 10, end=None)
-    fides_dir_name = ".fides"
-    fides_dir_path = f"{fides_directory_location}/{fides_dir_name}"
-    config_file_name = "fides.toml"
-    config_path = f"{fides_dir_path}/{config_file_name}"
     config = ctx.obj["CONFIG"]
 
-    included_values = {
-        "database": {
-            "server",
-            "user",
-            "password",
-            "port",
-            "db",
-            "test_db",
-        },
-        "logging": {
-            "level",
-            "destination",
-            "serialization",
-        },
-        "cli": {"server_protocol", "server_host", "server_port", "analytics_id"},
-        "user": {"analytics_opt_out"},
-    }
     click.echo(FIDES_ASCII_ART)
     click.echo("Initializing fides...")
-    separate()
+    print_divider()
 
-    # create the .fides dir if it doesn't exist
-    if not os.path.exists(fides_dir_path):
-        os.mkdir(fides_dir_path)
-        echo_green(f"Created a '{fides_dir_path}' directory.")
-    else:
-        click.echo(f"Directory '{fides_dir_path}' already exists.")
-
-    separate()
-
-    # create a fides.toml config file if it doesn't exist
-    if not os.path.isfile(config_path):
-        # request explicit consent for analytics collection
-        click.echo(OPT_OUT_COPY, nl=False)
-        config.user.analytics_opt_out = bool(input(OPT_OUT_PROMPT).lower() == "n")
-
-        separate()
-
-        with open(config_path, "w", encoding="utf-8") as config_file:
-            config_dict = config.dict(include=included_values)
-            toml.dump(config_dict, config_file)
-
-        echo_green(f"Created a fides config file: {config_path}")
-        click.echo("To learn more about configuring fides, see:")
-        click.echo("\thttps://ethyca.github.io/fides/installation/configuration/")
-
-    else:
-        click.echo(f"Configuration file already exists: {config_path}")
+    # create the config file as needed
+    config_path = create_config_file(
+        ctx=ctx, fides_directory_location=fides_directory_location
+    )
+    print_divider()
 
-    separate()
-    click.echo("For example policies and help getting started, see:")
-    click.echo("\thttps://ethyca.github.io/fides/guides/policies/")
-    separate()
+    # request explicit consent for analytics collection
+    check_and_update_analytics_config(ctx=ctx, config_path=config_path)
```

Comment:
Previously, `fides init` had it's own logic for collecting consent. I've now refactored it to use the same util function that the `with_analytics` helper does, so the prompts & logic is consistent.